packages/graphql-codegen-cli/src/utils/watcher.ts

```diff
@@ -84,13 +79,17 @@ export const createWatcher = (
       filename,
       config: normalizeOutputParam(config.generates[filename]),
     }))) {
+      // ParcelWatcher expects relative ignore patterns to be relative from watchDirectory,
+      // but we expect filename from config to be relative from cwd, so we need to convert
+      const filenameRelativeFromWatchDirectory = relative(watchDirectory, resolve(process.cwd(), entry.filename));
+
       if (entry.config.preset) {
         const extension = entry.config.presetConfig?.extension;
         if (extension) {
-          ignored.push(join(entry.filename, '**', '*' + extension));
+          ignored.push(join(filenameRelativeFromWatchDirectory, '**', '*' + extension));
         }
       } else {
-        ignored.push(entry.filename);
+        ignored.push(filenameRelativeFromWatchDirectory);
       }
     }
```

Comment:
This fixes a bug that would have been introduced in #9266 (although anyone it affected wouldn't even have been able to even get as far as noticing it was a bug prior to that fix).

Specifically: `ParcelWatcher.Options["ignore"]` expects that any relative paths or globs are relative from the `watchDirectory`, but we were passing them relative to `process.cwd()` (since that's how they're defined in the config). But after #9266, it's possible for `watchDirectory` to be something other than `process.cwd()`, so we need to convert these paths to be relative from `watchDirectory`.

I added (a lot of...) test code to make assertions on which paths and globs are passed to `ParcelWatcher.Options.ignore` and added regression tests for this.

Related (not to watch mode, but similar idea): #9272